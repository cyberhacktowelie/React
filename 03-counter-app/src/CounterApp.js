import { useState } from 'react';
import { PropTypes } from 'prop-types';
const CounterApp = ({value}) => {

    // const state = useState('Goku');
    // console.log(state);

    const [counter, setCounter] = useState(value);
    // console.log(counter, setCounter);

    //handleAdd
    const handleAdd = () => {
        // setCounter(counter + 1);
        setCounter(c => c + 1);
        // console.log('handleAdd');
    }

    const handleReset = () => {
        setCounter(() => value);
    }
    const handleSub = () => {
        setCounter(c => c - 1);
    }


    return (
        <>
            <h1> CounterApp </h1>
            <h2> {counter} </h2>

            {/* <button onClick={ (e) => { handleAdd() } } >+1</button> */}
            <button onClick={ handleAdd } >+1</button>
            <button onClick={ handleReset } >Reset</button>
            <button onClick={ handleSub } >-1</button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number
}

CounterApp.defaultProps = {
    value: 0
}

export default CounterApp;