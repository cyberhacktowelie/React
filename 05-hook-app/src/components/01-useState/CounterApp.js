import { useState } from "react";
import './counter.css';

export const CounterApp = () => {

    // const [ {count1,count2 }, setCount] = useState({
    //     count1: 0,
    //     count2: 20
    // });

    const [ state, setState] = useState({
        count1: 0,
        count2: 20,
        count3: 30,
        count4: 40
    });

    const {count1,count2} = state;

    return (
        <>
            <h1>Counter App</h1>
            <h2>Counter1 {count1}</h2>
            <h2>Counter2 {count2}</h2>
            <hr/>

            <button className="btn btn-primary" onClick={ () => {
                // setCount({count1:count1 + 1 , count2:count2});
                setState({...state ,count1:count1 + 1});
            }}>
                +1
            </button>    
        </>
    )
}