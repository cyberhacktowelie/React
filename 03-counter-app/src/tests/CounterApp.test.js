import CounterApp from "../CounterApp";
import { shallow } from "enzyme";


describe    ('Pruebas en el archivo CounterApp.test.js', () => {

    // let wrapper;//undefined
    let wrapper = shallow(<CounterApp />);//para no perder el intelisense lo inicializamos

    beforeEach(() => {
        wrapper = shallow(<CounterApp />);
    });

    test('<CounterApp /> debe mostrarse correctamente (match snapshot) y sus valores por defecto', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('CounterApp debe mostrar el valor por defecto 100', () => {
        const valorInicial = '100';
        const wrapper = shallow(
            <CounterApp 
            value={ 100 }/>);
        expect( wrapper ).toMatchSnapshot();

        const textoParrafo = wrapper.find('h2').text().trim();
        expect( textoParrafo ).toEqual( valorInicial );
    });

    test('CounterApp debe incrementar el valor en 1', () => {
        const incrementButton = wrapper.find('button').at(0);
        incrementButton.simulate('click');

        const textoParrafo = wrapper.find('h2').text().trim();
        expect( textoParrafo ).toEqual( '1' );
    });

    test('CounterApp debe decrementar el valor en 1', () => {
        // wrapper.setProps({ value: 0 });
        const decrementButton = wrapper.find('button').at(2);
        decrementButton.simulate('click');

        const textoParrafo = wrapper.find('h2').text().trim();
        expect( textoParrafo ).toEqual( '-1' );
    });

    test('CounterApp debe resetear correctamente al pulsar boton reset', () => {

        const wrapper = shallow(
            <CounterApp 
            value={ 105 }/>);
        const incrementButton = wrapper.find('button').at(0);
        incrementButton.simulate('click');
        incrementButton.simulate('click');
        const resetButton = wrapper.find('button').at(1);
        resetButton.simulate('click');
        const textoParrafo = wrapper.find('h2').text().trim();
        expect( textoParrafo ).toEqual( '105' );
    });

});