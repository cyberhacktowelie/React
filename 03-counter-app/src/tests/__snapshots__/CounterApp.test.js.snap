// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo CounterApp.test.js <CounterApp /> debe mostrarse correctamente (match snapshot) y sus valores por defecto 1`] = `
<Fragment>
  <h1>
     CounterApp 
  </h1>
  <h2>
     
    0
     
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;

exports[`Pruebas en el archivo CounterApp.test.js CounterApp debe mostrar el valor por defecto 100 1`] = `
<Fragment>
  <h1>
     CounterApp 
  </h1>
  <h2>
     
    100
     
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
